app/routes.py

```diff
@@ -1,2 +1,279 @@
+    videos_response = []
+    for video in videos:
+        videos_response.append(video.to_json())
+        
+    return jsonify(videos_response), 200
+
+@videos_bp.route("/<video_id>", methods=["GET"], strict_slashes=False)
+def get_video_by_id(video_id):
+    video = Video.query.get_or_404(video_id)
+
+    return video.to_json(), 200
+
+@videos_bp.route("", methods=["POST"], strict_slashes=False)
+def new_video():
+    request_body = request.get_json()
+    
+    try:
+        video = Video(title=request_body["title"],
+                        release_date=request_body["release_date"],
+                        total_inventory=request_body["total_inventory"],
+                        available_inventory=request_body["total_inventory"])
+
+    except KeyError:
+        return make_response({
+            "details": "Invalid data"
+        }, 400)
+
+    db.session.add(video)
+    db.session.commit()
+
+    return make_response({
+        "id": video.video_id
+    }, 201)
+
+@videos_bp.route("/<video_id>", methods=["PUT"], strict_slashes=False)
+def update_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    form_data = request.get_json()
+
+    try:
+        video.title = form_data["title"]
+        video.release_date = form_data["release_date"]
+        video.total_inventory = form_data["total_inventory"]
+
+    except KeyError:
+        return make_response({
+            "details": "Invalid data"
+        }, 400)
+
+    db.session.commit()
+
+    return video.to_json(), 200
+
+@videos_bp.route("/<video_id>", methods=["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    
+    db.session.delete(video)
+    db.session.commit()
+
+    return make_response({
+        "id": video.video_id
+    }, 200)
+
+@rentals_bp.route("/check-out", methods=["POST"], strict_slashes=False)
+def check_out():
+    request_body = request.get_json()
+
+    rental = Rental(customer_id=request_body["customer_id"],
+                        video_id=request_body["video_id"],
+                        due_date=datetime.datetime.now() + datetime.timedelta(days=7))
+
+    try:
+        customer = Customer.query.get(int(rental.customer_id))
+        video = Video.query.get(int(rental.video_id))
+
+        if customer == None:
+            return make_response({
+                "details": f"Customer {customer.customer_id} not found"
+            }, 404)
+    
+        elif video == None:
+            return make_response({
+                "details": f"Video {video.video_id} not found"
+            }, 404)
+    
+    except ValueError:
+        return make_response({
+            "details": "Invalid id"
+        }, 400) 
+
+    if video.available_inventory < 1:
+        return make_response({
+            "details": f"{video.title}: No available inventory"
+        }, 400)  
+        
+    else:
+        video.available_inventory -= 1
+        customer.videos_checked_out_count += 1
```

Comment:
If you compute the `available_inventory` and `videos_checked_out_count` from the rentals then you don't have to worry about managing these fields and they can't get out of sync.